src/cljs/athens/events.cljs

```diff
@@ -338,6 +338,20 @@
          :dispatch [:page/create title uid]}))))
 
 
+(reg-event-fx
+  :daily-note/delete
+  (fn [{:keys [db]} [_ uid title]]
+    (let [all-blocks         (db/retract-uid-recursively uid)
+          retract-blocks     (if (db/e-by-av :block/uid uid)  ;; If a page for the Daily Note exists
+                               all-blocks                 ;; retract both the page and the child blocks
+                               (rest all-blocks))         ;; else, only retract the child blocks
```

Comment:
I still don't understand how a date page have child blocks if the date page doesn't exist.

Reply:
Why not just dispatch `:page/delete` from `:daily-note/delete`? (Assuming tx-data looks the same)

i.e.

```
(reg-event-fx
  :daily-note/delete
  (fn [{:keys [db]} [_ uid title]]
    (let [filtered-dn        (filterv #(not= % uid) (:daily-notes/items db)) ;; Filter current date from daily note vec
          new-db (assoc db :daily-notes/items filtered-dn)]
      {:fx [[:dispatch [:page/delete uid title]]]
       :db new-db})))
```

Reply:
Ok. So I figured out why that error was popping up. When I just dispatched the `:page/delete` event directly from the button click. the daily note would be retracted, but it was still in the daily notes vector. The code here does prevent that from happening. Updating the PR now.

> (reg-event-fx
>   :daily-note/delete
>   (fn [{:keys [db]} [_ uid title]]
>     (let [filtered-dn        (filterv #(not= % uid) (:daily-notes/items db)) ;; Filter current date from daily note vec
>           new-db (assoc db :daily-notes/items filtered-dn)]
>       {:fx [[:dispatch [:page/delete uid title]]]
>        :db new-db})))
> ```